flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/writer/RecordWriter.java

```diff
@@ -108,15 +108,25 @@ private void sendToTarget(T record, int targetChannel) throws IOException, Inter
 
 		synchronized (serializer) {
 			SerializationResult result = serializer.addRecord(record);
+
 			while (result.isFullBuffer()) {
 				Buffer buffer = serializer.getCurrentBuffer();
 
 				if (buffer != null) {
-					writeBuffer(buffer, targetChannel, serializer);
+					writeAndClearBuffer(buffer, targetChannel, serializer);
+
+					// If this was a full record, we are done. Not breaking
+					// out of the loop at this point will lead to another
+					// buffer request before breaking out (that would not be
+					// a problem per se, but it can lead to stalls in the
+					// pipeline).
+					if (result.isFullRecord()) {
+						break;
```

Comment:
Can this only happen when the end of the record and the end of the buffer are exactly aligned?